# Introduction

- <a href= '#understanding'>Understanding the Dataset</a>
- <a href= '#normality'>Normality Test</a>
  - <a href= '#shapiro-wilk'>Shapiro-Wilk Test</a>
- <a href= '#anova'>Analysis of Variance (ANOVA)</a>
  - <a href= '#one-way-anova'>One-way ANOVA</a>
    - <a href= '#one-way-anova-levene'>Levene's Test</a>
    - <a href= '#one-way-anova-descr'>Descriptive Statistics</a>
    - <a href= '#one-way-anova-table'>ANOVA Table</a>
  - <a href= '#two-way-anova'>Two-way ANOVA</a>
    - <a href= '#two-way-anova-levene'>Levene's Test</a>
    - <a href= '#two-way-anova-descr'>Descriptive Statistics</a>
    - <a href= '#two-way-anova-table'>ANOVA Table</a>
    - <a href= '#two-way-anova-tukey'>Tukey Test</a>
- <a href= '#manova'>Multivariate Analysis of Variance (MANOVA)</a>
  - <a href= '#one-way-manova'>One-way MANOVA</a>
    - <a href= '#one-way-manova-boxm'>Homogeneity of Covariance Matrices</a>
    - <a href= '#one-way-manova-levene'>Levene's Test</a>
    - <a href= '#one-way-manova-descr'>Descriptive Statistics</a>
    - <a href= '#one-way-manova-table'>MANOVA Table</a>
  - <a href= '#two-way-manova'>Two-way MANOVA</a>
    - <a href= '#two-way-manova-boxm'>Homogeneity of Covariance Matrices</a>
    - <a href= '#two-way-manova-descr'>Descriptive Statistics</a>
    - <a href= '#two-way-manova-table'>MANOVA Table</a>

<a id='understanding'></a>

# Understanding the Dataset


In [1]:
# R Packages
packages <- c("tidyverse", "ggplot2", "rstatix", "pander", "heplots", "readr")

suppressPackageStartupMessages(lapply(packages, require, character.only = TRUE))

# Importing Data
df <- read.csv("./R/dataset/myopia-study.csv", sep = ",")

# for kaggle
# df <- read.csv("../input/myopia-study/myopia.csv", sep = ";")

# Top 6 rows
head(df)

[[1]]
[1] TRUE

[[2]]
[1] TRUE

[[3]]
[1] TRUE

[[4]]
[1] TRUE

[[5]]
[1] TRUE

[[6]]
[1] TRUE

ID STUDYYEAR MYOPIC AGE GENDER SPHEQ  AL    ACD   LT    VCD   SPORTHR READHR
1 1  1992      1      6   1      -0.052 21.89 3.690 3.498 14.70 45       8    
2 2  1995      0      6   1       0.608 22.38 3.702 3.392 15.29  4       0    
3 3  1991      0      6   1       1.179 22.49 3.462 3.514 15.52 14       0    
4 4  1990      1      6   1       0.525 22.20 3.862 3.612 14.73 18      11    
5 5  1995      0      5   0       0.697 23.29 3.676 3.454 16.16 14       0    
6 6  1995      0      6   0       1.744 22.14 3.224 3.556 15.36 10       6    
  COMPHR STUDYHR TVHR DIOPTERHR MOMMY DADMY
1 0      0       10   34        1     1    
2 1      1        7   12        1     1    
3 2      0       10   14        0     0    
4 0      0        4   37        0     1    
5 0      0        4    4        1     0    
6 2      1       19   44        0     1

In [2]:
class(df)


[1] "data.frame"

In [3]:
print(sapply(df, class))


       ID STUDYYEAR    MYOPIC       AGE    GENDER     SPHEQ        AL       ACD 
"integer" "integer" "integer" "integer" "integer" "numeric" "numeric" "numeric" 
       LT       VCD   SPORTHR    READHR    COMPHR   STUDYHR      TVHR DIOPTERHR 
"numeric" "numeric" "integer" "integer" "integer" "integer" "integer" "integer" 
    MOMMY     DADMY 
"integer" "integer" 


In [4]:
# Structure of the data frame
str(df)


'data.frame':	618 obs. of  18 variables:
 $ ID       : int  1 2 3 4 5 6 7 8 9 10 ...
 $ STUDYYEAR: int  1992 1995 1991 1990 1995 1995 1993 1991 1991 1991 ...
 $ MYOPIC   : int  1 0 0 1 0 0 0 0 0 0 ...
 $ AGE      : int  6 6 6 6 5 6 6 6 7 6 ...
 $ GENDER   : int  1 1 1 1 0 0 1 1 0 1 ...
 $ SPHEQ    : num  -0.052 0.608 1.179 0.525 0.697 ...
 $ AL       : num  21.9 22.4 22.5 22.2 23.3 ...
 $ ACD      : num  3.69 3.7 3.46 3.86 3.68 ...
 $ LT       : num  3.5 3.39 3.51 3.61 3.45 ...
 $ VCD      : num  14.7 15.3 15.5 14.7 16.2 ...
 $ SPORTHR  : int  45 4 14 18 14 10 12 12 4 30 ...
 $ READHR   : int  8 0 0 11 0 6 7 0 0 5 ...
 $ COMPHR   : int  0 1 2 0 0 2 2 0 3 1 ...
 $ STUDYHR  : int  0 1 0 0 0 1 1 0 1 0 ...
 $ TVHR     : int  10 7 10 4 4 19 8 8 3 10 ...
 $ DIOPTERHR: int  34 12 14 37 4 44 36 8 12 27 ...
 $ MOMMY    : int  1 1 0 0 1 0 0 0 0 0 ...
 $ DADMY    : int  1 1 0 1 0 1 1 0 0 0 ...


In [5]:
# Statistical summary
summary(df)


       ID          STUDYYEAR        MYOPIC            AGE       
 Min.   :  1.0   Min.   :1990   Min.   :0.0000   Min.   :5.000  
 1st Qu.:155.2   1st Qu.:1991   1st Qu.:0.0000   1st Qu.:6.000  
 Median :309.5   Median :1992   Median :0.0000   Median :6.000  
 Mean   :309.5   Mean   :1992   Mean   :0.1311   Mean   :6.299  
 3rd Qu.:463.8   3rd Qu.:1994   3rd Qu.:0.0000   3rd Qu.:6.000  
 Max.   :618.0   Max.   :1995   Max.   :1.0000   Max.   :9.000  
     GENDER           SPHEQ               AL             ACD       
 Min.   :0.0000   Min.   :-0.6990   Min.   :19.90   Min.   :2.772  
 1st Qu.:0.0000   1st Qu.: 0.4562   1st Qu.:22.04   1st Qu.:3.424  
 Median :0.0000   Median : 0.7290   Median :22.46   Median :3.585  
 Mean   :0.4887   Mean   : 0.8010   Mean   :22.50   Mean   :3.579  
 3rd Qu.:1.0000   3rd Qu.: 1.0340   3rd Qu.:22.97   3rd Qu.:3.730  
 Max.   :1.0000   Max.   : 4.3720   Max.   :24.56   Max.   :4.250  
       LT             VCD           SPORTHR          READHR      
 Mi

In [6]:
# Check missing values
sum(is.na(df))


[1] 0

<a id='normality'></a>

# Normality Test

The assumption of normality should be considered to perform the analysis.
Considering the normality of all variables;

- H0: Variables provide the assumption of normality

- H1: Variables do not satisfy the assumption of normality

</br>

- p ≤ (α = 0.05) : **reject** H0

- p \> (α = 0.05) : reject H1 (or **accept** H0) [[ref]](https://www.geeksforgeeks.org/p-value/)

<a id='shapiro-wilk'></a>

## Shapiro-Wilk Test

Shapiro-Wilk's test is a normality test in frequentist statistics.

It is among the three tests for normality designed for detecting all kinds of departure from normality [[ref]](https://www.geeksforgeeks.org/shapiro-wilk-test-in-r-programming/).


In [7]:
results_01 <- lapply(df, shapiro.test)
shapiro_data <- as.data.frame(do.call(rbind, results_01))
shapiro_data <- shapiro_data[-c(1), ]

# p-value check

p <- c(as.numeric(shapiro_data$p.value))


shapiro_data$p.value <- scales::pvalue(p,
  accuracy = 0.05,
  decimal.mark = ".",
  add_p = TRUE
)

shapiro_data %>%
  filter(!is.na(as.numeric(shapiro_data$statistic))) %>%
  select(!data.name)


statistic p.value method                     
STUDYYEAR 0.8988441 p<0.05  Shapiro-Wilk normality test
MYOPIC    0.3972498 p<0.05  Shapiro-Wilk normality test
AGE       0.6476244 p<0.05  Shapiro-Wilk normality test
GENDER    0.6363741 p<0.05  Shapiro-Wilk normality test
SPHEQ     0.892849  p<0.05  Shapiro-Wilk normality test
AL        0.9974372 p=0.45  Shapiro-Wilk normality test
ACD       0.9980938 p=0.75  Shapiro-Wilk normality test
LT        0.9968676 p=0.30  Shapiro-Wilk normality test
VCD       0.9963458 p=0.15  Shapiro-Wilk normality test
SPORTHR   0.9309436 p<0.05  Shapiro-Wilk normality test
READHR    0.8322593 p<0.05  Shapiro-Wilk normality test
COMPHR    0.6702958 p<0.05  Shapiro-Wilk normality test
STUDYHR   0.6734526 p<0.05  Shapiro-Wilk normality test
TVHR      0.9321961 p<0.05  Shapiro-Wilk normality test
DIOPTERHR 0.9029071 p<0.05  Shapiro-Wilk normality test
MOMMY     0.6364998 p<0.05  Shapiro-Wilk normality test
DADMY     0.636557  p<0.05  Shapiro-Wilk normality test

<u>**Comment**</u><br>

Since the p values of the Axial Length (AL), Anterior Chamber Depth (ACD), Lens Thickness (LT) and Vitreous Chamber Depth (VCD) variables are greater than 0.05, these variables provide the assumption of normality.

The GENDER variable returns 0 = Male and 1 = Female, but these should not be considered numeric as they are just levels.
In order for such variables to be treated as factors rather than numbers, we need to explicitly convert them to factors using the `as.factor()` function.


In [8]:
df$AGE <- as.factor(df$AGE)
df$MYOPIC <- as.factor(df$MYOPIC)
df$GENDER <- as.factor(df$GENDER)
df$MOMMY <- as.factor(df$MOMMY)
df$DADMY <- as.factor(df$DADMY)


<a id='anova'></a>

# Analysis of Variance (ANOVA)

ANOVA is a statistical method that separates observed variance data into different components to use for additional tests [[ref]](https://www.investopedia.com/terms/a/anova.asp).

**_A one-way ANOVA uses one independent variable, while a two-way ANOVA uses two independent variables_** [[ref]](https://www.scribbr.com/statistics/one-way-anova/).

<a id='one-way-anova'></a>

## One-way ANOVA

> Dependent Variable: Lens Thickness (LT)
>
> Independent Variable ("Factor"): Gender

<a id='one-way-anova-levene'></a>

### Levene's Test

Levene's test is an inferential statistic used to evaluate the **equality of variances** for a variable determined for two or more groups [[ref]](https://www.geeksforgeeks.org/levenes-test-in-r-programming/).

For Equality of Variance,

- H0: Variances are homogeneous.
- H1: The variance is not homogeneous.


In [9]:
# examine whether it has equal variance

print("AL - Axial Length")
levene_test(data = df, AL ~ GENDER, center = mean)

print("ACD - Anterior Chamber Depth")
levene_test(data = df, ACD ~ GENDER, center = mean)

print("LT - Lens Thickness")
levene_test(data = df, LT ~ GENDER, center = mean)

print("VCD - Vitreous Chamber Depth")
levene_test(data = df, VCD ~ GENDER, center = mean)


[1] "AL - Axial Length"


df1 df2 statistic p        
1 1   616 0.1506335 0.6980648

[1] "ACD - Anterior Chamber Depth"


df1 df2 statistic p       
1 1   616 1.866422  0.172384

[1] "LT - Lens Thickness"


df1 df2 statistic p        
1 1   616 0.2940678 0.5878216

[1] "VCD - Vitreous Chamber Depth"


df1 df2 statistic p        
1 1   616 0.1236948 0.7251815

<u>**Comment**</u><br>

Since p values are greater than 0.05, the H0 hypothesis is accepted variances can be considered homogeneous.

<a id='one-way-anova-descr'></a>

### Descriptive Statistics


In [10]:
group_by(df, GENDER) %>%
  summarise(
    n = n(),
    mean = mean(LT, na.rm = TRUE),
    sd = sd(LT, na.rm = TRUE),
    se = sd / sqrt(n),
    LCL = mean - qt(1 - (0.05 / 2), n - 1) * se,
    UCL = mean + qt(1 - (0.05 / 2), n - 1) * se,
    min = min(LT, na.rm = TRUE),
    max = max(LT, na.rm = TRUE)
  )


GENDER n   mean     sd        se          LCL      UCL      min   max  
1 0      316 3.530152 0.1510087 0.008494901 3.513438 3.546866 3.154 3.940
2 1      302 3.553278 0.1574916 0.009062622 3.535444 3.571112 2.960 4.112

For Output;

- n : The number of observations for each gender.

- mean : The mean value for each gender.

- median : The median value for each gender.

- sd : Standard Deviation

- se : Standard Error

- LCL, UCL : It means the lower and upper confidence intervals of the mean. Assuming that the data are normally distributed, we can be 95% sure that the true mean is between the lower and upper values indicated for each treatment group.

- min, max : The minimum and maximum value for each gender.

<a id='one-way-anova-table'></a>

### ANOVA Table


In [11]:
model_a1 <- aov(LT ~ GENDER, data = df)

pander(summary(model_a1))




|    &nbsp;     | Df  | Sum Sq  | Mean Sq | F value | Pr(>F)  |
|:-------------:|:---:|:-------:|:-------:|:-------:|:-------:|
|  **GENDER**   |  1  | 0.08259 | 0.08259 |  3.473  | 0.06286 |
| **Residuals** | 616 |  14.65  | 0.02378 |   NA    |   NA    |

Table: Analysis of Variance Model



In the ANOVA Table, it was tested whether there was a difference in lens thickness between the groups.

- H0: There is no difference

- H1: There is difference

<u>**Comment**</u><br>

This output indicates whether the results between groups are significant.
The columns for the F value and Pr(\>F) in the table correspond to the p-value.
We know that H0 will be rejected if it is less than 0.05.

Pr(\>F) value is 0.0629.
In this case, since the thickness of the lens is greater than 0.05, it can be considered that there is no difference.

<u>**NOTE**</u><br>
For _IBM SPSS Statistics_ : `Analyze > Compare Means > One-Way ANOVA`. You can perform the analysis by ticking the ones you want in the "Options" section.

<a id='two-way-anova'></a>

## Two-way ANOVA

> Dependent Variables: Spherical Equivalent Refraction (SPHEQ)
>
> Independent Variables ("Factors"): GENDER, AGE

<a id='two-way-anova-levene'></a>

### Levene's Test


In [12]:
levene_test(data = df, SPHEQ ~ GENDER * AGE, center = mean)


df1 df2 statistic p        
1 9   608 0.6516839 0.7527991

<u>**Comment**</u><br>

Since the p-values are greater than 0.05, the H0 hypothesis is <u>accepted</u> and it can be said that the variances are homogeneous.

<a id='two-way-anova-descr'></a>

### Descriptive Statistics


In [13]:
group_by(df, GENDER, AGE) %>%
  summarise(
    mean = mean(SPHEQ, na.rm = TRUE),
    sd = sd(SPHEQ, na.rm = TRUE),
    n = n(), .groups = "drop"
  )


GENDER AGE mean       sd        n  
1  0      5    0.4418000 0.3006795  10
2  0      6    0.8398419 0.6180858 215
3  0      7    0.8413333 0.6389336  63
4  0      8    0.4080435 0.3184835  23
5  0      9   -0.1046000 0.5504101   5
6  1      5    0.6217273 0.5195924  11
7  1      6    0.8558548 0.6275674 241
8  1      7    0.7808947 0.5519041  19
9  1      8    0.6289667 0.7435502  30
10 1      9    1.3680000        NA   1

<a id='two-way-anova-table'></a>

### ANOVA Table


In [14]:
model_a2 <- lm(
  SPHEQ ~ GENDER * AGE,
  data = df,
  contrasts = list(GENDER = contr.sum, AGE = contr.sum)
)

two_way <- anova_test(model_a2,
  type = 3,
  detailed = TRUE
)

pander(two_way)




|   Effect    |  SSn  |  SSd  | DFn | DFd |   F   |    p     | p<.05 |  ges  |
|:-----------:|:-----:|:-----:|:---:|:---:|:-----:|:--------:|:-----:|:-----:|
| (Intercept) | 28.9  | 230.1 |  1  | 608 | 76.37 | 2.28e-17 |   *   | 0.112 |
|   GENDER    | 2.165 | 230.1 |  1  | 608 | 5.722 |  0.017   |   *   | 0.009 |
|     AGE     | 6.772 | 230.1 |  4  | 608 | 4.474 |  0.001   |   *   | 0.029 |
| GENDER:AGE  | 2.452 | 230.1 |  4  | 608 | 1.62  |  0.168   |       | 0.011 |



For Output;

Table: Tests of Between-Subjects Effects

- SSn: Type III Sum of Squares
- DFn: df, degrees of freedom
- p: Sig., significance level
- ges: Partial Eta Squared

<u>**Comment**</u><br>

This output indicates whether the independent categorical variables or their interactions are statistically significant.
Here we should look at the p-values (GENDER, AGE, GENDER:AGE).

- The p-value for gender is 0.0170 ((p ≤ 0.05) : **reject** H0).

There is a difference in Spherical Equivalent Refraction (SPHEQ) level between men and women.

- The p-value for age is 0.0014 ((p ≤ 0.05) : **reject** H0).

There is a difference in the Spherical Equivalent Refraction (SPHEQ) level between the ages.

- The p-value for the interaction between gender and age is 0.1676.

<u>**NOTE**</u><br>

For _IBM SPSS Statistics_ : `Analyze > General Linear Model > Univariate`. You can perform the analysis by ticking the ones you want in the "Options" section.

<a id='two-way-anova-tukey'></a>

### Tukey Test

The results of the test for homogeneity of variance (Levene's Test) showed that they had equal variances. In the later ANOVA test, H0 was rejected. For this reason, it is necessary to look at the results of the Tukey HSD test.

The column mean difference shows the difference between the means of the two groups being compared.


In [15]:
tukey_hsd(model_a2, which = "AGE")


term group1 group2 null.value estimate     conf.low    conf.high    p.adj  
1  AGE  5      6      0           0.312066429 -0.06355348  0.687686338 0.15500
2  AGE  5      7      0           0.303140218 -0.10846751  0.714747948 0.26000
3  AGE  5      8      0          -0.004667632 -0.43862769  0.429292428 1.00000
4  AGE  5      9      0          -0.380715241 -1.15978813  0.398357650 0.66800
5  AGE  6      7      0          -0.008926211 -0.21080155  0.192949130 1.00000
6  AGE  6      8      0          -0.316734061 -0.56097588 -0.072492246 0.00382
7  AGE  6      9      0          -0.692781669 -1.38436485 -0.001198486 0.04940
8  AGE  7      8      0          -0.307807850 -0.60442995 -0.011185749 0.03750
9  AGE  7      9      0          -0.683855458 -1.39562642  0.027915506 0.06650
10 AGE  8      9      0          -0.376047609 -1.10097403  0.348878813 0.61500
   p.adj.signif
1  ns          
2  ns          
3  ns          
4  ns          
5  ns          
6  **          
7  *           
8  *           
9  ns          
10 ns

<u>**Comment**</u><br>

Column Sig. (p-value) indicates whether the mean difference is statistically significant. There is a difference between 6-8, 6-9 and 7-8 years old.

<u>**NOTE**</u><br>

For _IBM SPSS Statistics_ : `Analyze > General Linear Model > Univariate`. You can perform the analysis by ticking the ones you want in the "Post Hoc" section.

<a id='manova'></a>

# Multivariate Analysis of Variance (MANOVA)

Multivariate analysis of variance (MANOVA) is an extension of univariate analysis of variance (ANOVA) that measures the effect of independent categorical variables on a large number of dependent continuous variables.

One-way MANOVA is used to determine whether there is a difference between groups (two or more) of the categorical variables on more than one continuous dependent variable.

Two-way MANOVA is used to determine whether there is an interaction between two independent categorical variables on two or more continuous dependent variables.

<u>**Summary**</u><br>

> **One-way ANOVA**
> 1 Independent Variable, 1 Dependent Variable

> **Two-way ANOVA**
> 2 Independent Variables, 1 Dependent

> **One-way MANOVA**
> 1 Independent Variable, 1+ Dependent Variables

> **Two-way MANOVA**
> 2 Independent Variables, 2+ Dependent Variables

<a id='one-way-manova'></a>

## One-way MANOVA

> Dependent Variables: Anterior Chamber Depth (ACD), Axial Length (AL)
>
> Independent Variable ("Factor"): AGE

<a id='one-way-manova-boxm'></a>

### Homogeneity of Covariance Matrices

It is an assumption that must be met in multivariate analysis. Homogeneity of covariance is when multiple groups have equal covariance matrices in an experimental design or statistical test.


In [16]:
boxM(cbind(ACD, AL) ~ AGE, data = df)



	Box's M-test for Homogeneity of Covariance Matrices

data:  Y
Chi-Sq (approx.) = 8.82, df = 12, p-value = 0.7182


<u>**Comment**</u><br>

Among the covariance matrices between dependent variables;

- H0: There is no difference
- H1: There is a difference.

Since the P-value is greater than 0.05, the H0 hypothesis is accepted. It provides equality of covariance.

<a id='one-way-manova-levene'></a>

### Levene's Test


In [17]:
print("ACD - Anterior Chamber Depth")
levene_test(data = df, ACD ~ AGE, center = mean)

print("AL - Axial Length")
levene_test(data = df, AL ~ AGE, center = mean)


[1] "ACD - Anterior Chamber Depth"


df1 df2 statistic p      
1 4   613 0.6344901 0.63805

[1] "AL - Axial Length"


df1 df2 statistic p        
1 4   613 0.718439  0.5795184

<u>**Comment**</u><br>

H0 hypothesis is accepted because p-values are greater than 0.05. The assumption of equality of variance is also provided.

<a id='one-way-manova-descr'></a>

### Descriptive Statistics


In [18]:
df %>%
  group_by(AGE) %>%
  get_summary_stats(ACD, AL, type = "mean_sd")


AGE variable n   mean   sd   
1  5   ACD       21  3.518 0.244
2  5   AL        21 22.280 0.602
3  6   ACD      456  3.557 0.227
4  6   AL       456 22.426 0.675
5  7   ACD       82  3.640 0.211
6  7   AL        82 22.707 0.653
7  8   ACD       53  3.671 0.221
8  8   AL        53 22.783 0.576
9  9   ACD        6  3.763 0.364
10 9   AL         6 23.273 0.937

<a id='one-way-manova-table'></a>

### MANOVA Table


In [19]:
model_m1 <- lm(cbind(ACD, AL) ~ AGE,
  data = df
)

# Multivariate Tests
summary(Manova(model_m1,
  type = 3
))



Type III MANOVA Tests:

Sum of squares and products for error:
         ACD        AL
ACD 31.49637  39.82391
AL  39.82391 270.55319

------------------------------------------
 
Term: (Intercept) 

Sum of squares and products for the hypothesis:
          ACD        AL
ACD  259.8324  1645.814
AL  1645.8142 10424.812

Multivariate Tests: (Intercept)
                 Df test stat approx F num Df den Df     Pr(>F)    
Pillai            1   0.97473 11804.83      2    612 < 2.22e-16 ***
Wilks             1   0.02527 11804.83      2    612 < 2.22e-16 ***
Hotelling-Lawley  1  38.57786 11804.83      2    612 < 2.22e-16 ***
Roy               1  38.57786 11804.83      2    612 < 2.22e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

------------------------------------------
 
Term: AGE 

Sum of squares and products for the hypothesis:
         ACD        AL
ACD 1.254779  4.292674
AL  4.292674 14.866303

Multivariate Tests: AGE
                 Df test stat  approx F n

<u>**Comment**</u><br>

Since p-value is smaller than 0.05, it can be said that there is a difference between Anterior Chamber Depth (ACD) and Axial Length (AL) groups.


In [20]:
summary.aov(model_m1)


 Response ACD :
             Df  Sum Sq  Mean Sq F value    Pr(>F)    
AGE           4  1.2548 0.313695  6.1053 8.039e-05 ***
Residuals   613 31.4964 0.051381                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

 Response AL :
             Df  Sum Sq Mean Sq F value    Pr(>F)    
AGE           4  14.866  3.7166  8.4208 1.286e-06 ***
Residuals   613 270.553  0.4414                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1


<a id='two-way-manova'></a>

## Two-way MANOVA

> Dependent Variables: DIOPTERHR, TVHR, SPORTHR
>
> Independent Variables ("Factors"): GENDER, MYOPIC

<a id='two-way-manova-boxm'></a>

### Homogeneity of Covariance Matrices


In [21]:
boxM(cbind(DIOPTERHR, TVHR, SPORTHR) ~ GENDER * MYOPIC, data = df)



	Box's M-test for Homogeneity of Covariance Matrices

data:  Y
Chi-Sq (approx.) = 21.77, df = 18, p-value = 0.2423


<a id='two-way-manova-descr'></a>

### Descriptive Statistics


In [22]:
df %>%
  group_by(GENDER, MYOPIC) %>%
  get_summary_stats(DIOPTERHR, TVHR, SPORTHR, type = "mean_sd")


MYOPIC GENDER variable  n   mean   sd    
1  0      0      DIOPTERHR 281 26.747 16.684
2  0      0      TVHR      281  9.199  6.180
3  0      0      SPORTHR   281 13.203  8.095
4  1      0      DIOPTERHR  35 26.914 16.655
5  1      0      TVHR       35  8.371  5.281
6  1      0      SPORTHR    35  9.171  7.501
7  0      1      DIOPTERHR 256 24.734 15.009
8  0      1      TVHR      256  8.691  5.328
9  0      1      SPORTHR   256 11.219  7.618
10 1      1      DIOPTERHR  46 28.022 16.986
11 1      1      TVHR       46  9.283  5.265
12 1      1      SPORTHR    46 10.522  8.398

<a id='two-way-manova-table'></a>

### MANOVA Table


In [23]:
model_m2 <-
  lm(cbind(DIOPTERHR, TVHR, SPORTHR) ~ GENDER * MYOPIC, data = df)

# Multivariate Tests
summary(Manova(model_m2,
  type = 3
))



Type III MANOVA Tests:

Sum of squares and products for error:
          DIOPTERHR      TVHR   SPORTHR
DIOPTERHR 157794.72 23932.997 14647.270
TVHR       23933.00 20128.958  4706.911
SPORTHR    14647.27  4706.911 38233.637

------------------------------------------
 
Term: (Intercept) 

Sum of squares and products for the hypothesis:
          DIOPTERHR     TVHR  SPORTHR
DIOPTERHR 201032.94 69141.85 99232.60
TVHR       69141.85 23780.16 34129.36
SPORTHR    99232.60 34129.36 48982.56

Multivariate Tests: (Intercept)
                 Df test stat approx F num Df den Df     Pr(>F)    
Pillai            1 0.7131297 507.1227      3    612 < 2.22e-16 ***
Wilks             1 0.2868703 507.1227      3    612 < 2.22e-16 ***
Hotelling-Lawley  1 2.4858956 507.1227      3    612 < 2.22e-16 ***
Roy               1 2.4858956 507.1227      3    612 < 2.22e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

------------------------------------------
 
Term: GENDER 

Sum of sq

In [24]:
summary.aov(model_m2)


 Response DIOPTERHR :
               Df Sum Sq Mean Sq F value Pr(>F)
GENDER          1    362  361.83  1.4079 0.2359
MYOPIC          1    254  253.73  0.9873 0.3208
GENDER:MYOPIC   1    169  168.53  0.6558 0.4184
Residuals     614 157795  256.99               

 Response TVHR :
               Df  Sum Sq Mean Sq F value Pr(>F)
GENDER          1    16.4  16.425  0.5010 0.4793
MYOPIC          1     0.1   0.105  0.0032 0.9549
GENDER:MYOPIC   1    34.9  34.855  1.0632 0.3029
Residuals     614 20129.0  32.783               

 Response SPORTHR :
               Df Sum Sq Mean Sq F value   Pr(>F)   
GENDER          1    417  417.23  6.7004 0.009868 **
MYOPIC          1    332  332.33  5.3370 0.021208 * 
GENDER:MYOPIC   1    192  192.44  3.0904 0.079253 . 
Residuals     614  38234   62.27                    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1


<u>**NOTE**</u><br>

For _IBM SPSS Statistics_ : `Analyze > General Linear Model > Multivariate`. You can perform the analysis by ticking the ones you want in the "Options" section.
